.github/workflows/test.yml

```diff
@@ -50,7 +53,26 @@ jobs:
         run: go vet ./...
 
       - name: Test
-        run: go test -race -coverprofile=coverage.out -coverpkg=./... ./...
+        env:
+          TIGER_PUBLIC_KEY_INTEGRATION: ${{ secrets.TIGER_PUBLIC_KEY_INTEGRATION }}
+          TIGER_SECRET_KEY_INTEGRATION: ${{ secrets.TIGER_SECRET_KEY_INTEGRATION }}
+          TIGER_PROJECT_ID_INTEGRATION: ${{ secrets.TIGER_PROJECT_ID_INTEGRATION }}
+        run: |
+          # Run all tests with race detection and coverage
+          # Integration tests will skip automatically if secrets are not set
+          go test -race -coverprofile=coverage.out -coverpkg=./... ./...
```

Comment:
But the secrets will be set from the env: section above. So this will run all integration tests. I don't think you need any of the code below


This is the skip logic btw 
```
	publicKey := os.Getenv("TIGER_PUBLIC_KEY_INTEGRATION")
	secretKey := os.Getenv("TIGER_SECRET_KEY_INTEGRATION")
	projectID := os.Getenv("TIGER_PROJECT_ID_INTEGRATION")


	if publicKey == "" || secretKey == "" || projectID == "" {
		t.Skip("Skipping service not found test: TIGER_PUBLIC_KEY_INTEGRATION, TIGER_SECRET_KEY_INTEGRATION, and TIGER_PROJECT_ID_INTEGRATION must be set")
	}```

so just setting the env is enough. We don't need the .env file or script